addons/product_visible_discount/product_visible_discount.py

```diff
@@ -21,19 +19,17 @@ class sale_order_line(osv.osv):
     def get_real_price_currency(self, cr, uid, product_id, res_dict, qty, uom, pricelist, context=None):
         """Retrieve the price before applying the pricelist"""
         item_obj = self.pool['product.pricelist.item']
-        price_type_obj = self.pool['product.price.type']
         product_obj = self.pool['product.product']
         field_name = 'list_price'
+        currency_id = None
         if res_dict.get(pricelist):
             rule_id = res_dict[pricelist][1]
         else:
             rule_id = False
         if rule_id:
             item_base = item_obj.browse(cr, uid, rule_id, context=context).base
-            if item_base > 0:
-                price_type = price_type_obj.browse(cr, uid, item_base)
-                field_name = price_type.field
-                currency_id = price_type.currency_id
+            field_name = 'list_price' if item_base == 'list_price' else 'standard_price'
```

Comment:
and also use uom._compute_price instead of uom._compute_qty to convert price into final uom.